private/buf/buffetch/internal/internal.go

```diff
@@ -704,18 +766,23 @@ type GetBucketOption func(*getBucketOptions)
 
 // WithGetBucketTerminateFileNames only applies if subdir is specified.
 //
+// The terminate files are organised as a slice of slices of file names.
+// Priority will be given to the first slice of terminate file names, and then
+// the next, etc. This allows the `internal` layer to process the terminate file
+// hierarchies without dependencies on the origin of the termiante files.
```

Comment:
We reference the `buf.work.yaml` file in other comments to help make this readable, so feel free to comment with concrete examples rather than being restricted by the generic nature of `buffetch`. See `WithGetBucketTerminateFileNames` for reference.